src/mtype.c

```diff
@@ -5635,8 +5635,8 @@ Type *TypeFunction::semantic(Loc loc, Scope *sc)
                 }
                 else
                 {
-                    error(loc, "auto can only be used for template function parameters");
-                    errors = true;
+                    fparam->storageClass &= ~STCauto;
+                    fparam->storageClass |= STCrvref;
```

Comment:
Currently applying unitary `auto` attribute to function parameter is forbidden. So adding new `auto ref` feature should not change compiler behavior to accept currently invalid code without good reason.